src/atoms/visual-story/cover-page/cover-page.tsx

```diff
@@ -18,16 +18,34 @@ export const CoverPageBase = ({ story, config }: CoverPageProps) => {
-      ? visualStoriesConfig.logoAlignment(config)
-      : visualStoriesConfig.logoAlignment;
-  const logoUrl =
-    typeof visualStoriesConfig.logoUrl === "function"
-      ? visualStoriesConfig.logoUrl(config)
-      : visualStoriesConfig.logoUrl;
-  const visualStoryConfig = { logoUrl, logoAlignment };
+  let customTemplatelogoAlignment = get(config, ["opts", "featureConfig", "visualStories", "logoAlignment"], null);
+  let customTemplatelogoUrl = get(config, ["opts", "featureConfig", "visualStories", "logoUrl"], null);
+
+  const visualStoriesConfigWithThemes = get(config, ["opts", "featureConfig", "visualStories"]) || [];
+  
+  if (Array.isArray(visualStoriesConfigWithThemes)) {
+    const visualStoryTheme = get(story, ["metadata", "story-attributes", "visualstorytheme"]) || [];
+    const theme = visualStoryTheme[0];
+    switch (theme) {
+      case "theme-2":
+        customTemplatelogoAlignment = visualStoriesConfigWithThemes[1] && (typeof visualStoriesConfigWithThemes[1].logoAlignment === "function" ? visualStoriesConfigWithThemes[1].logoAlignment(config) : visualStoriesConfigWithThemes[1].logoAlignment);
+        customTemplatelogoUrl = visualStoriesConfigWithThemes[1] && (typeof visualStoriesConfigWithThemes[1].logoUrl === "function" ? visualStoriesConfigWithThemes[1].logoUrlThemeBased(config) : visualStoriesConfigWithThemes[1].logoUrl);
+        break;
+      case "theme-3":
+        customTemplatelogoAlignment = visualStoriesConfigWithThemes[2] && (typeof visualStoriesConfigWithThemes[2].logoAlignment === "function" ? visualStoriesConfigWithThemes[2].logoAlignment(config) : visualStoriesConfigWithThemes[2].logoAlignment);
+        customTemplatelogoUrl = visualStoriesConfigWithThemes[2] && (typeof visualStoriesConfigWithThemes[2].logoUrl === "function" ? visualStoriesConfigWithThemes[2].logoUrlThemeBased(config) : visualStoriesConfigWithThemes[2].logoUrl);
+        break;
+      default:
+        customTemplatelogoAlignment = visualStoriesConfigWithThemes[0] && (typeof visualStoriesConfigWithThemes[0].logoAlignment === "function" ? visualStoriesConfigWithThemes[0].logoAlignment(config) : visualStoriesConfigWithThemes[0].logoAlignment);
+        customTemplatelogoUrl = visualStoriesConfigWithThemes[0] && (typeof visualStoriesConfigWithThemes[0].logoUrl === "function" ? visualStoriesConfigWithThemes[0].logoUrlThemeBased(config) : visualStoriesConfigWithThemes[0].logoUrl);
+    }
+  }
+
+  const visualStoriesConfig = get(config, ["opts", "featureConfig", "visualStories"]) || {};
+  const logoAlignment = typeof visualStoriesConfig.logoAlignment === "function" ? visualStoriesConfig.logoAlignment(config) : visualStoriesConfig.logoAlignment;
+  const logoUrl = typeof visualStoriesConfig.logoUrl === "function" ? visualStoriesConfig.logoUrl(config) : visualStoriesConfig.logoUrl;
+
+  const visualStoryConfig = { logoUrl, logoAlignment, customTemplatelogoAlignment , customTemplatelogoUrl };
```

Comment:
don't pass customTemplatelogoAlignment to child component. The decision to use which logo url should be done here itself. Clean up this code a little, there's some repetition

```js
const visualStoriesConfig = get(config, ["opts", "featureConfig", "visualStories"]) || {};
const visualStoriesConfigWithThemes = get(config, ["opts", "featureConfig", "visualStories"]) || [];
```

Reply:
Done